src/jvmMain/kotlin/com/sdercolin/vlabeler/repository/SampleInfoRepository.kt

```diff
@@ -44,11 +44,13 @@ object SampleInfoRepository {
     suspend fun load(project: Project, sampleFile: File, moduleName: String, appConf: AppConf): Result<SampleInfo> {
         val maxSampleRate = appConf.painter.amplitude.resampleDownToHz
         val normalize = appConf.painter.amplitude.normalize
+        val mergeChannels = appConf.painter.power.mergeChannels
         infoMap[moduleName to sampleFile.name]?.let {
             if (it.maxSampleRate == maxSampleRate &&
                 it.normalize == normalize &&
                 it.getFile(project).exists() &&
-                it.lastModified == sampleFile.lastModified()
+                it.lastModified == sampleFile.lastModified() &&
+                ((mergeChannels && it.powerChannels == 1) || (!mergeChannels && it.channels == it.powerChannels))
```

Comment:
Probably we will need to check `hasPower` and `hasSpectrogram` as well?
We can have a member function in `SampleInfo` class to handle this logic. 
```kotlin
fun shouldReload(appConf: AppConf, algorithmVersion: Int, ...): Boolean = ...
```